pkg/skaffold/runner/dev.go

```diff
@@ -151,14 +151,14 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 					return build.DependenciesForArtifact(ctx, artifact, r.runCtx)
 				},
 				func(e filemon.Events) {
-					s, err := sync.NewItem(ctx, artifact, e, r.builds, r.runCtx)
+					s, err := sync.NewItem(ctx, artifact, e, r.builds, r.runCtx, len(g[artifact.ImageName]))
 					switch {
 					case err != nil:
 						logrus.Warnf("error adding dirty artifact to changeset: %s", err.Error())
 					case s != nil:
 						r.changeSet.AddResync(s)
 					default:
-						r.changeSet.AddRebuild(artifact)
+						addRebuild(g, artifact, r.changeSet.AddRebuild, r.runCtx.Opts.IsTargetImage)
```

Comment:
can we just call `r.runCtx.Opts.IsTargetImage(artifact)` and pass that boolean as a parameter here rather than passing the function?

Reply:
ah actually i just realized that we're passing the rebuild function as well which gets called on the same artifact param, so we may as well pass them both as functions.